kocherga/kocherga_serial.hpp

```diff
@@ -416,26 +381,33 @@ template <typename Callback>
     {
         ok = ok && out(x);
     }
-    for (const auto x : crc.getBytes())
+    for (const auto x : UserData)
     {
         ok = ok && out(x);
     }
-    crc = {};  // Now it's the payload CRC.
+    const auto headerBytes = headerCrc.getBytes();
```

Comment:
```suggestion
    const auto header_crc_bytes = header_crc.getBytes();
```

1. Naming convention (please update elsewhere)
2. `header_crc_bytes` is a better name. Also why couldn't you use range-based for here?

Reply:
The header CRC is big endian and I wasn't aware of another way of reverse looping before c++20. Or should I switch out getBytes on CRC16 to emit in opposite byte order?

Reply:
I'll just switch the byte order in CRC16CCITT.

Reply:
Yes indeed, it makes sense that `getBytes` returns the bytes in the order that is native to the implemented CRC algorithm.

Reply:
Done. Will push shortly